File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        
        if(x < 0) return false;
        int num = x;
        int ans = 0;

        while(num) {
            if(ans > INT_MAX/10) return false;
            ans = (ans*10) + (num%10);
            num/=10;
        }

        return ans == x;
    }
};

File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        
        if(x < 0) return false;
        return x == reverseNumber(x);
    }

    private int reverseNumber(int num) {
        int res = 0;

        while(num != 0) {
            res = (res*10) + (num%10);
            num/=10;
        }

        return res;
    }
}